sessions/2018/october/Session_15.md

```diff
@@ -0,0 +1,324 @@
+# SOFTWARE ENGINEERING IN JAVA
+
+## Session 15 (16/10/2018)
+
+- Reviewing/fixing Spring API homework
+- API desing concepts
+- Dependency injection concepts
+
+### Notes:
+
+Roger was reviewing my homework solution and advicing me some modifications and improvements.
+
+#### Reviewing/fixing Spring API homework
+
+In certain part of the code I needed to save and incremental `id` `int` field into the object's field that was a `String`. A way to do it should be changing the type of the object's field, but a simpler way (hack) to work around it is doing `id` increment and mapping to `String` by concatenating with `""` at the moment to set into the object's field:
```

Comment:
Crucially, managing and deciding on ID values should be done on the backend (ideally by the database itself), and never by the client.